File: app/src/main/java/com/example/moneymanager/GamingRoomActivity.java
package com.example.moneymanager;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

public class GamingRoomActivity extends AppCompatActivity {
    ArrayList<ActivityList> activityList = new ArrayList<>();

    int activity_logos[] = {R.drawable.ic_shopping_cart, R.drawable.withdrawal_icon,
            R.drawable.gaming_icon};
    String activity_names[] = {"Shopping", "Withdrawal", "Gaming"};
    String activity_dates[] = {"Jan 24, 2022", "Jan 23, 2022", "Jan 21, 2022"};
    String activity_balance[] = {"-$54", "-$54", "-$54"};

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_gaming_room);

        createActivityList();

        RecyclerView activityRecycler = findViewById(R.id.rvActivities);
        activityRecycler.setLayoutManager(new LinearLayoutManager(this, RecyclerView.VERTICAL,
                false));
        GamingRoomAdapter firstSection = new GamingRoomAdapter(activityList);
        activityRecycler.setAdapter(firstSection);
    }

    public void createActivityList() {
        int count = 0;
        for (String names: activity_names) {
            ActivityList activityList = new ActivityList(activity_logos[count], names,
                    activity_dates[count], activity_balance[count]);
            this.activityList.add(activityList);
            count++;
        }
    }
}

File: app/src/main/java/com/example/moneymanager/GamingRoomActivity.kt
package com.example.moneymanager

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.example.moneymanager.databinding.ActivityGamingRoomBinding
import com.example.moneymanager.model.Activity

private lateinit var binding:ActivityGamingRoomBinding

class GamingRoomActivity : AppCompatActivity() {
    private val activities = listOf(
        Activity(R.drawable.ic_shopping_cart, "Shopping", "Jan 24, 2022", "$-54"),
        Activity(R.drawable.withdrawal_icon, "Withdrawal", "Jan 23, 2022", "$-54"),
        Activity(R.drawable.gaming_icon, "Gaming", "Jan 21, 2022", "$-54")
    )

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityGamingRoomBinding.inflate(layoutInflater)
        setContentView(binding.root)

         binding.rvActivities.adapter =  GamingRoomAdapter(activities)

        binding.btnBack.setOnClickListener {
            val intent = Intent(this@GamingRoomActivity, MainActivity::class.java)
            startActivity(intent) }
        }
}


